Beta Version 0.5

ElykDeer opened the pull request:
Still Does Not Have a Graphical API, but the framework is almost entirely there! So, why not push version 0.5?  Gotta figure out licensing before I can start making the graphical API, and thus version 1.0.

ElykDeer committed:
Calculates Hexmap Size

Segfaults when run, but the map size should be correctly calculated. (Segfaults due to insertions/deletions)

ElykDeer committed:
Updated Help Menu

Removed tick marks from next to "save" and "compile"

ElykDeer committed:
Corrected Seperate Compilation

There was some code in the .h I moved to the .cpp

ElykDeer committed:
Created Hex Class

The data structure will soon hold hexes instead of sets of Entity pointers, this is the first step to get there..

ElykDeer committed:
Test With More Objects

ElykDeer committed:
Started Conversion To Hex

Hex coordinates are almost being calculated.. The framework is there, and it's being filled in slowly.

ElykDeer committed:
Can Now Save/Get Col/Row

ElykDeer committed:
Const And Unsigned Correctness

Added some consts and unsigned's around

ElykDeer committed:
hexCord Renamed hexOffsetCord

There will be another function to get the HexCoordinates of a hex, called hexCord later

ElykDeer committed:
Implimented Guess/Check Algorithm

Segfaults on run, still need to setup what it checks against

ElykDeer committed:
Hexmap Draft One

byLocal is broken, runtime segfaults (on insertions)
Will fix later.

ElykDeer committed:
Added Error Checking/Debugging Code

ElykDeer committed:
More Debug Code

The segfault always happens in the same hex

ElykDeer committed:
Hexmap Draft One Done

FInished converting to hexmap - byLocal still broken

ElykDeer committed:
Merge pull request #6 from SaplingStudios/hexmap

Hexmap Draft One

ElykDeer committed:
Hexmap Wrap Around

Coordinates inserted in the map now, if they were to fall of the map, instead wrap around as if the hexagon map was a simple projection of a real world

ElykDeer committed:
Compiler V2

Seperated color stuffs and added command tab-completion

ElykDeer committed:
Iterate Over Everything

getAll now provides an iterator so one can use a ranged for over everything in the bin

ElykDeer committed:
Updated Readme

Added dependancy

ElykDeer committed:
Update All

Can now update all entities in the dataStructure

ElykDeer committed:
Sorta Fixed byLocal

So far only generates an interator for the entities (of a certain type or of all things) that are within the same hex - need to expand this to allow for custom distances, also need to add size tracking, also need to fix compilation when including Hex.cpp in the depends.txt

ElykDeer committed:
Reorganization

Sepeated the code for the dataStructure, to make it easier to mantain in the future.

ElykDeer committed:
More Code Factorization

Turned the bit that converts from cartesian coordinates to hexagonal coordinates into a subclass of the data structure

ElykDeer committed:
Modified Hex File Structure

Hex is now a derived class from "HexInternals" so that the "Hex" class can be moved elsewhere and included back into this file.  If moved into the "Plugins" folder, then the Hex class as defined can be used as a template for enviorntments that people make for themselves.

ElykDeer committed:
Added Hex Iter

Added a hex iter for convience, though I realized while making it that the user cannot use the iterator and I might as well just toss this bit, even though it's convient - it's not critical in any manner

ElykDeer committed:
Added move() And updateHexes()

Also changed a variable from "bins" to "hexes"

ElykDeer committed:
Cleaned Up/Added Test Code

Added timers for things! Yayyyyyyyy!!

ElykDeer committed:
Updated Test Code

Just made for some nicer output

ElykDeer committed:
Command Prompt Word Change

The command promt's prompt was "compile," which was entirely a mistake.. I just changed it to "command"

ElykDeer committed:
Added The 'Near' Class Of Iterators

It. Finally.  Works.

ElykDeer committed:
Merge remote-tracking branch 'origin/unstable-kyle' into unstable-kyle

ElykDeer committed:
Fixing Merge Error

Forgot to define a variable.. "theTime"

ElykDeer committed:
Updated GitIgnore

Finally removed compiler and the main program from tracking.

ElykDeer committed:
Removed datTest.cpp From Main/

There's a backup in Plugins/datTest/

ElykDeer committed:
Moved File

ElykDeer committed:
Added Resolution

The resolution represents how many seconds in game go by in one tick.  A res of 1 means that (1 tick == 1 game sec), a res of 2 means that (1 tick == 2 game secs)

ElykDeer committed:
Updated GitIgnore

Was ignoring too much!

ElykDeer committed:
Added MultiThreading

ElykDeer committed:
Added Comments And Fixed Tick Count

The tick count ought to change based on the resolution of simulation... so it keeps track of how many game ticks (game seconds) have gone by.

ElykDeer committed:
Tracks Entity Count

Each hex and the overall structure tracks how many entities it holds.

KyleMiles commented:
Need to calculate pixel-distance to hex-distance.

KyleMiles commented:
Need to separate the Hex class from the file with the hex internals... put in the plugins folder?

ElykDeer committed:
Added Functions To ThreadManager

Added wait(), sleep(), pause(), and resume()

ElykDeer committed:
Cleaner Memory Allocation, "Duration"

Allocate all the entities near each other, seems to improve the speed of this... But it's inconsistant.  Will remove the poorly made "duration" thing later

ElykDeer committed:
Anded "lasTime"

Remove this godawful implimentation later, please...

ElykDeer committed:
Changed Main And Compiler Program Names

Changed them to avoid more issues.

ElykDeer committed:
Fix Makefile

ElykDeer committed:
Added Linux Option To The Makefile

ElykDeer committed:
Added The Wrong File Last Time

ElykDeer committed:
Adjust gitIgnore For Commit Four Back

Fixing for commit: b359751f2570774f43a5fbe4623d4b736e1914bc

ElykDeer committed:
Moved Hex To Enviornment (In Plugins)

So every plugin set needs its own enviornment to live in, and that will be developed over in the plugin area, in an "Enviornment" folder.  Also, adjusted the compiler to ignore the new folder in "Plugins"

ElykDeer committed:
First Draft Of Distance Calculation

getAllOfTypeNear and getAllNear now have a semi-real pix-distance to hex-distance calculation scheme.  Will probably be changed later.

KyleMiles commented:
Update Readme!! Remove debug code later..
